src/crypto/keccak.c

```diff
@@ -104,10 +104,22 @@ void keccak(const uint8_t *in, size_t inlen, uint8_t *md, int mdlen)
     
     memset(st, 0, sizeof(st));
 
-    for ( ; inlen >= rsiz; inlen -= rsiz, in += rsiz) {
+    if (((uintptr_t)in) & 7) { // 64 bit aligned or not ?
+      for ( ; inlen >= rsiz; inlen -= rsiz, in += rsiz) {
+        for (i = 0; i < rsizw; i++) {
+          uint64_t ina;
+          memcpy(&ina, &((uint64_t*)in)[i], 8);
```

Comment:
shouldn't the source pointer be (void *)? otherwise the compiler will still assume uint64_t alignment, no?

Reply:
OK, I also added a better unit test for this. I can't seem to be able to connect to ssh for now, I'll push later when I can.

Reply:
Tested and implemented. This will be successful